rules/onchain_budget.go

```diff
@@ -278,20 +304,33 @@ func (o *OnChainBudgetEnforcer) checkers() map[string]mid.RoundTripChecker {
 				for _, c := range r.Channels {
 					totalAmt += c.LocalFundingAmount +
 						c.PushSat
+					c.Memo = fmt.Sprintf(
+						"%s:%s", id, c.Memo,
+					)
 				}
 
-				return o.handlePendingPayment(
+				err = o.handlePendingPayment(
 					ctx, &onChainAction{
 						Amount:     uint64(totalAmt),
 						ActionType: actionTypeOpenChannel,
 					},
 				)
```

Comment:
good catch :pray:! I haven't spent the time to fully flesh that commit out because it was still uncertain if we will use the identifier, but I should have done that

Reply:
I added a check in the unit tests and the itest also has a check for the memo.